File: src-tauri/test_api.py
import os
import json
import urllib.request
import urllib.error
import glob

def test_fetch_models():
    print("--- 测试 fetchAvailableModels ---")
    data_dir = os.path.join(os.path.expanduser("~"), ".antigravity_cockpit", "accounts")
    files = glob.glob(os.path.join(data_dir, "*.json"))
    
    if not files:
        print("没有找到账号文件")
        return None
        
    # 寻找 chenyiding01@gmail.com 账号
    account = None
    for f in files:
        try:
            with open(f, 'r', encoding='utf-8') as file:
                acc = json.load(file)
                if acc.get("email") == "chenyiding01@gmail.com":
                    account = acc
                    break
        except Exception:
            pass
            
    if not account:
        print("找不到指定的账号 chenyiding01@gmail.com")
        return None
        
    print(f"使用账号: {account['email']}")
    
    access_token = account.get("token", {}).get("access_token")
    project_id = account.get("token", {}).get("project_id", "")
    
    body = {}
    data = json.dumps(body).encode('utf-8')
    
    req = urllib.request.Request(
        "https://cloudcode-pa.googleapis.com/v1internal:fetchAvailableModels",
        data=data,
        headers={
            "Authorization": f"Bearer {access_token}",
            "Content-Type": "application/json",
            "User-Agent": "grpc-java-okhttp/1.68.2"
        },
        method="POST"
    )
    
    try:
        # 使用系统默认代理
        proxy_handler = urllib.request.ProxyHandler()
        opener = urllib.request.build_opener(proxy_handler)
        response = opener.open(req, timeout=15)
        
        print(f"fetchAvailableModels HTTP Status: {response.getcode()}")
        resp_data = json.loads(response.read().decode('utf-8'))
        
        models = list(resp_data.get("models", {}).keys())
        if models:
            print(f"\n============================")
            print(f"成功获取到 {len(models)} 个模型!")
            print(f"前 5 个模型: {', '.join(models[:5])}")
            print(f"============================\n")
            return {"account": account, "models": models}
        else:
            print("响应中没有 models 字段:", resp_data)
            
    except urllib.error.HTTPError as e:
        print(f"HTTPError: {e.code}")
        print(e.read().decode('utf-8'))
    except Exception as e:
        print(f"请求失败: {e}")
        
    return None

def test_chat_proxy(model):
    print(f"\n--- 测试 Chat 代理请求 (model: {model}) ---")
    data = json.dumps({
        "model": model,
        "max_tokens": 100,
        "stream": True,
        "messages": [{"role": "user", "content": "测试消息，只需回复 hello"}]
    }).encode('utf-8')
    
    # 获取 IDE 设置中当前的代理端口，这里直接盲猜默认的 19530 或 8045，用户截图里有 19530
    req = urllib.request.Request(
        "http://127.0.0.1:19530/antigravity/v1/messages",
        data=data,
        headers={
            "Content-Type": "application/json",
            "x-api-key": "chat-test",
            "anthropic-version": "2023-06-01"
        },
        method="POST"
    )
    
    try:
        # 本地请求不使用代理
        proxy_handler = urllib.request.ProxyHandler({})
        opener = urllib.request.build_opener(proxy_handler)
        response = opener.open(req, timeout=15)
        
        print(f"Chat 代理 HTTP Status: {response.getcode()}")
        for line in response:
            line_str = line.decode('utf-8').strip()
            if line_str:
                print(f"STREAM CHUNK: {line_str}")
    except urllib.error.HTTPError as e:
        print(f"HTTPError: {e.code}")
        print(e.read().decode('utf-8'))
    except Exception as e:
        print(f"请求失败: {e}")

if __name__ == "__main__":
    result = test_fetch_models()
    if result and result["models"]:
        test_chat_proxy(result["models"][0])

File: src-tauri/test_api.mjs
import fs from 'fs';
import path from 'path';
import os from 'os';
import { fetch as undiciFetch } from 'undici';

// 如果有本地代理，请取消注释并按需修改
// import { ProxyAgent, setGlobalDispatcher } from 'undici';
// const proxyAgent = new ProxyAgent('http://127.0.0.1:7890');
// setGlobalDispatcher(proxyAgent);

async function testFetchModels() {
    console.log('--- 测试 fetchAvailableModels ---');
    const dataDir = path.join(os.homedir(), '.antigravity_cockpit', 'accounts');
    const files = fs.readdirSync(dataDir).filter(f => f.endsWith('.json'));

    if (files.length === 0) {
        console.log('没有找到账号文件');
        return;
    }

    // 寻找 chenyiding01@gmail.com 账号
    let account = null;
    for (const file of files) {
        const acc = JSON.parse(fs.readFileSync(path.join(dataDir, file), 'utf8'));
        if (acc.email === 'chenyiding01@gmail.com') {
            account = acc;
            break;
        }
    }

    if (!account) {
        console.log('找不到指定的账号 chenyiding01@gmail.com');
        return;
    }

    console.log(`使用账号: ${account.email}`);

    const accessToken = account.token.access_token;
    const projectId = account.token.project_id || '';

    console.log(`access_token length: ${accessToken.length}, project_id: ${projectId}`);

    try {
        const body = projectId ? { project: projectId } : {};
        // 增加 fetch 错误捕捉
        console.log('开始请求 Google endpoint...');
        const response = await undiciFetch('https://cloudcode-pa.googleapis.com/v1internal:fetchAvailableModels', {
            method: 'POST',
            headers: {
                'Authorization': `Bearer ${accessToken}`,
                'Content-Type': 'application/json',
                'User-Agent': 'grpc-java-okhttp/1.68.2'
            },
            body: JSON.stringify(body),
        });

        console.log(`fetchAvailableModels HTTP Status: ${response.status}`);
        const data = await response.json();

        if (response.status !== 200) {
            console.log('Error data:', JSON.stringify(data, null, 2));
        }

        if (data.models) {
            const models = Object.keys(data.models);
            console.log(`\n============================`);
            console.log(`成功获取到 ${models.length} 个模型!`);
            console.log(`前 5 个模型: ${models.slice(0, 5).join(', ')}`);
            console.log(`============================\n`);
            return { account, models };
        } else {
            console.log('响应中没有 models 字段:', data);
        }
    } catch (e) {
        console.error('请求失败:', e);
    }
    return null;
}

async function main() {
    const result = await testFetchModels();

    if (result && result.models.length > 0) {
        const testModel = result.models[0];
        console.log(`\n--- 测试 Chat 代理请求 (model: ${testModel}) ---`);
        try {
            // 请在此处修改你的代理端口（默认可能是19530）
            const response = await undiciFetch('http://127.0.0.1:19530/antigravity/v1/messages', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'x-api-key': 'chat-test',
                    'anthropic-version': '2023-06-01'
                },
                body: JSON.stringify({
                    model: testModel,
                    max_tokens: 100,
                    stream: true,
                    messages: [{ role: 'user', content: '测试消息，只需回复 hello' }]
                })
            });

            console.log(`Chat 代理 HTTP Status: ${response.status}`);
            if (!response.ok) {
                console.log('Chat 代理返回错误:', await response.text());
                return;
            }

            const reader = response.body.getReader();
            const decoder = new TextDecoder();
            while (true) {
                const { done, value } = await reader.read();
                if (done) break;
                const chunk = decoder.decode(value);
                console.log('STREAM CHUNK:', chunk);
            }
        } catch (e) {
            console.error('代理请求失败:', e);
        }
    }
}

main();
